<a href="https://colab.research.google.com/github/MileneCV/project_2/blob/main/project2_part1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Milene Carmes Vallejo
10/31/2022

# First choice: Stroke Prediction Dataset 

In [1]:
import pandas as pd 
from google.colab import drive
drive.mount('/content/drive')
filename = '/content/drive/My Drive/stroke_prediction_dataset.csv'
df = pd.read_csv(filename)

Mounted at /content/drive


In [3]:
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [4]:
df.shape

(5110, 12)

In [5]:
df.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

In [6]:
df.duplicated().sum()

0

##1. Source of data
https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset?select=healthcare-dataset-stroke-data.csv



##2. Brief description of data
This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status.


##3. What is the target?
stroke: 1 if the patient had a stroke or 0 if not

## 4. What does one row represent? (A person?  A business?  An event? A product?)
a person

## 5. Is this a classification or regression problem?
classification

##6. How many features does the data have?
12

##7. How many rows are in the dataset?
5110 

## 8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
201 missing values in bmi column

#Second choice: Heart Failure Predition Dataset



In [10]:
import pandas as pd 
from google.colab import drive
drive.mount('/content/drive')
filename = '/content/drive/My Drive/heart_prediction.csv'
df2 = pd.read_csv(filename)
df2.head()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [12]:
df2.isna().sum()

Age               0
Sex               0
ChestPainType     0
RestingBP         0
Cholesterol       0
FastingBS         0
RestingECG        0
MaxHR             0
ExerciseAngina    0
Oldpeak           0
ST_Slope          0
HeartDisease      0
dtype: int64

In [14]:
df2.shape

(918, 12)

##1. Source of data
https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

##2. Brief description of data
This dataset is used to predict whether a patient is likely to get Heart Failure based on the input parameters like gender, age, cholesterol.

##3. What is the target?
HeartDisease column

##4. What does one row represent? (A person? A business? An event? A product?)
Person

##5. Is this a classification or regression problem?
Classification


##6. How many features does the data have?
12
##7. How many rows are in the dataset?
918

##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
None